src/client/Microsoft.Identity.Client/Internal/Requests/UsernamePasswordRequest.cs

```diff
@@ -38,12 +41,31 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
         {
             await ResolveAuthorityAsync().ConfigureAwait(false);
             await UpdateUsernameAsync().ConfigureAwait(false);
-            var userAssertion = await FetchAssertionFromWsTrustAsync().ConfigureAwait(false);
-            var msalTokenResponse =
-                await SendTokenRequestAsync(GetAdditionalBodyParameters(userAssertion), cancellationToken).ConfigureAwait(false);
+
+            MsalTokenResponse msalTokenResponse = await GetTokenResponseAsync(cancellationToken).ConfigureAwait(false);
+            
             return await CacheTokenResponseAndCreateAuthenticationResultAsync(msalTokenResponse).ConfigureAwait(false);
         }
 
+        private async Task<MsalTokenResponse> GetTokenResponseAsync(CancellationToken cancellationToken)
+        {
+            if (_requestParameters.AppConfig.IsBrokerEnabled)
```

Comment:
You need to also check `broker.IsBrokerInstalledAndInvokable(_authenticationRequestParameters.AuthorityInfo.AuthorityType)`